src/kubernetes.cc

```diff
@@ -933,19 +933,16 @@ json::value KubernetesReader::GetOwner(
       std::vector<std::string>{api_version, kind, uid}, "/");
 
   std::lock_guard<std::recursive_mutex> lock(mutex_);
-  auto found = owners_.emplace(std::piecewise_construct,
-                               std::forward_as_tuple(encoded_ref),
-                               std::forward_as_tuple());
-  json::value& owner = found.first->second;
-  if (found.second) {  // Not found, inserted new.
+  auto found = owners_.find(encoded_ref);
+  if (found == owners_.end()) {  // Not found, add new.
```

Comment:
If the owner is found, let's return the owner right away.

Reply:
Tackled this another way by going back to a single return. PTAL.